TigranKirakosov/index.js

```diff
@@ -0,0 +1,253 @@
+  DOM = UICtrl.getDOMStrings();
+
+  Task = function(id, text, deadline) {
+    this.id = id;
+    this.text = text;
+    this.deadline = deadline;
+    this.isDone = 'undone';
+    this.iconClass = DOM.uncheckedBtn;
+  };
+  Task.prototype.toggleState = function() {
+    if (this.isDone === 'undone') {
+      this.isDone = 'done';
+      this.iconClass = DOM.checkedBtn;
+    } else if (this.isDone === 'done') {
+      this.isDone = 'undone';
+      this.iconClass = DOM.uncheckedBtn;
+    }
+  };
+  if (
+    localStorage.getItem('tasks') === null ||
+    localStorage.getItem('tasks') === ''
+  ) {
+    tasks = [];
+  } else {
+    tasks = JSON.parse(localStorage.getItem('tasks'));
+  }
+  return {
+    getFromLS: function() {
+      return tasks;
+    },
+    addTask: function(text, deadline) {
+      var newTask, ID;
+      if (tasks.length > 0) ID = tasks[tasks.length - 1].id + 1;
+      else ID = 0;
+      newTask = new Task(ID, text, deadline);
+      return newTask;
+    },
+    addToLS: function(task) {
+      tasks.push(task);
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+    },
+    deleteFromLS: function(id) {
+      var ids, index;
+      ids = tasks.map(function(current) {
+        return current.id;
+      });
+      index = ids.indexOf(id);
+      if (index !== -1) {
+        tasks.splice(index, 1);
+      }
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+    },
+    toggleTask: function(id) {
+      var ids, index;
+      ids = tasks.map(function(current) {
+        return current.id;
+      });
+      index = ids.indexOf(id);
+      if (index !== -1) {
+        Task.prototype.toggleState.call(tasks[index]);
+      }
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+    },
+    clearLS: function() {
+      tasks = [];
+      localStorage.setItem('tasks', []);
+    }
+  };
+})(UIController);
+
+var controller = (function(UICtrl, storageCtrl) {
+  var DOM = UICtrl.getDOMStrings();
+  var setUpEventListeners = function() {
+    document.addEventListener('DOMContentLoaded', ctrlDisplayTasks);
+    document
+      .querySelector(DOM.submitBtn)
+      .addEventListener('click', ctrlAddTask);
+    document.addEventListener('keypress', function(event) {
+      if (event.keyCode === 13 || event.which === 13) {
+        ctrlAddTask();
+        event.preventDefault();
+      }
+    });
+    document
+      .querySelector(DOM.taskList)
+      .addEventListener('click', ctrlDeleteTask);
+    document
+      .querySelector(DOM.taskList)
+      .addEventListener('click', ctrlToggleTask);
+    document
+      .querySelector(DOM.filterBtn)
+      .addEventListener('click', ctrlFilterTasks);
+    document
+      .querySelector(DOM.clearBtn)
+      .addEventListener('click', ctrlClearTasks);
+  };
+  var ctrlDisplayTasks = function() {
+    var tasks, taskList;
+    var html = '';
+    tasks = storageCtrl.getFromLS();
+    taskList = document.querySelector(DOM.taskList);
+    tasks.forEach(function(task) {

```

Comment:
use just `renderTasks` function that I mentioned in comment above here instead of code duplication